src/core/wasm-module.js

```diff
@@ -0,0 +1,164 @@
+class Impl {
+    static modules = {};
+
+    // returns true if the running host supports wasm modules (all browsers except IE)
+    static wasmSupported() {
+        let supported = null;
+
+        if (supported === null) {
+            supported = (() => {
+                try {
+                    if (typeof WebAssembly === "object" && typeof WebAssembly.instantiate === "function") {
+                        const module = new WebAssembly.Module(Uint8Array.of(0x0, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00));
+                        if (module instanceof WebAssembly.Module)
+                            return new WebAssembly.Instance(module) instanceof WebAssembly.Instance;
+                    }
+                } catch (e) { }
+                return false;
+            })();
+        }
+
+        return supported;
+    }
+
+    // load a script
+    static loadScript(url, callback) {
+        const s = document.createElement('script');
+        s.setAttribute('src', url);
+        s.onload = () => {
+            callback(null);
+        };
+        s.onerror = () => {
+            callback(`Failed to load script='${url}'`);
+        };
+        document.body.appendChild(s);
+    }
+
+    // load a wasm module
+    static loadWasm(moduleName, config, callback) {
+        const loadUrl = (Impl.wasmSupported() && config.glueUrl && config.wasmUrl) ? config.glueUrl : config.fallbackUrl;
+        if (loadUrl) {
+            Impl.loadScript(loadUrl, (err) => {
+                if (err) {
+                    callback(err, null);
+                } else {
+                    window[moduleName]({
+                        locateFile: () => config.wasmUrl,
+                        onAbort: () => {
+                            callback('wasm module aborted.');
+                        }
+                    }).then((instance) => {
+                        callback(null, instance);
+                    });
+                }
+            });
+        } else {
+            callback('No supported wasm modules found.', null);
```

Comment:
- could this add the module name to the error message
- and perhaps even log a debug warning - to help find the problem in case the user does not handle errors correctly

Reply:
The default error handler logs the details as you describe. Otherwise if the caller registers an error handler, it's up to them to do this.